sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/MobileServiceSyncContext.cs

```diff
@@ -370,6 +372,17 @@ private Task ExecuteOperationAsync(MobileServiceTableOperation operation, JObjec
 
                 try
                 {
+                    if (MobileServiceTableOperationKind.Insert != operation.Kind)
+                    {
+                        // Refresh the item's version, in case it changed while we were awaiting the item lock
+                        var localStoreItem = await this.Store.LookupAsync(operation.TableName, operation.ItemId);
```

Comment:
This is essentially enforcing a last write wins policy, we can't assume the pending item on the client can just overwrite the server's item without a conflict.  The server could easily have changed additional fields/etc that the prev client object may now accidently stomp on.

I can see the usefulness of this option, based on the specific app/table, but not generally.